src/main/webapp/app/pages/curation/GeneListPage.tsx

```diff
@@ -67,7 +68,26 @@ const GeneListPage = (props: IGeneListPage) => {
       accessor: 'hugoSymbol',
       Header: 'Hugo Symbol',
       Cell(cell: { value: string }): any {
-        return <Link to={getCurationPageLink(cell.value, isGermline)}>{cell.value}</Link>;
+        const link = getCurationPageLink(cell.value, isGermline);
+
+        return (
+          <Link
+            to={undefined}
+            onClick={async event => {
+              event.preventDefault();
+              if (props.firebaseDb) {
+                try {
+                  await createGeneIfDoesNotExist(cell.value, isGermline, props.firebaseDb, props.createGene);
+                  window.location.href = link;
```

Comment:
Do you think using `useHistory` and `history.push()` would be better? It should be faster because it is not doing a full page reload. For instance, I believe the navigation sidebar should be rendered pretty quickly. Worth a shot to see if a there's a difference.

Reply:
Good point, definitely feels better that way.